### PR TITLE
Fixing regression in census routing rules

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
Fixes https://github.com/ONSdigital/eq-survey-runner/issues/750
Also added tests to check routing rule

### CODE REVIEW

File: tests/functional/spec/census/issues/routing-last-week-were-you.spec.js [hunk 1]
```diff
@@ -88,4 +89,98 @@ describe('ArriveInUk', function () {
...
+    PastUsualAddress.clickPastUsualAddressAnswerThisAddress().submit()
+    Passports.clickPassportsAnswerUnitedKingdom().submit()
+    Disability.clickDisabilityAnswerNo().submit()
+    Qualifications.clickQualificationsEnglandAnswerUndergraduateDegree().submit()
+    Volunteering.clickVolunteeringAnswerNo().submit()
+    EmploymentType.clickEmploymentTypeAnswerWorkingAsAnEmployee().clickEmploymentTypeAnswerOnAGovernmentSponsoredTrainingScheme().clickEmploymentTypeAnswerNoneOfTheAbove().submit()
+    expect(Jobseeker.isOpen()).to.be.false
+  })
+
+  it('Given I am answering question 28.	Last week were you.., When I select multiple answers excluding none of the above, Then I am routed to 29. Were you actively looking for...?', function () {
```
Comment (ajmaddaford) [inline tests/functional/spec/census/issues/routing-last-week-were-you.spec.js]: Should this say '..Then I am routed to 35. In your main job, are (were) you?'

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

**Comment by ajmaddaford** [comment]
@yoganandkunche I thought the original intent was that 'None of the above' overrode the other options, so that if it was ticked, regardless of the other options, we shouldn't skip the 5 questions.

**Comment by collisdigital** [comment]
Please double check against the slides... I'll do the same

**Comment by collisdigital** [comment]
> if any of options 1-6 are selected next 5 questions are skipped irrespective of whether option 7 is selected. No options selected continue to next question. 

Next Slide:
> Only asked if 7th answer option is the only option none of hte options 1-6 were selected at question 28. 

6 Slides later:
> Asked if any of options 1-6 selected at question 28 or if ‘yes’ selected at question 33.

**Comment by collisdigital** [comment]
Therefore:
- If ANY of option 1-6 selected, ignore option 7 and goto Question 35 (skips 5 blocks)
- If ONLY option 7 (none of the above) OR no option selected: goto Question 29 (next block)

**Comment by jonnyshaw89** [comment]
@collisdigital what about if options 1,2 and 7 are selected?

**Comment by collisdigital** [comment]
Updated my comment above ^^

**Comment by jonnyshaw89** [comment]
The tests in this PR match the scenario @collisdigital has illustrated.

File: tests/functional/spec/census/issues/routing-last-week-were-you.spec.js [hunk 2]
```diff
@@ -88,4 +89,98 @@ describe('ArriveInUk', function () {
...
+    PastUsualAddress.clickPastUsualAddressAnswerThisAddress().submit()
+    Passports.clickPassportsAnswerUnitedKingdom().submit()
+    Disability.clickDisabilityAnswerNo().submit()
+    Qualifications.clickQualificationsEnglandAnswerUndergraduateDegree().submit()
+    Volunteering.clickVolunteeringAnswerNo().submit()
+    EmploymentType.clickEmploymentTypeAnswerWorkingAsAnEmployee().clickEmploymentTypeAnswerOnAGovernmentSponsoredTrainingScheme().clickEmploymentTypeAnswerNoneOfTheAbove().submit()
+    expect(Jobseeker.isOpen()).to.be.false
+  })
+
+  it('Given I am answering question 28.	Last week were you.., When I select multiple answers excluding none of the above, Then I am routed to 35. In your main job, are (were) you?', function () {
```
Comment (ajmaddaford) [inline tests/functional/spec/census/issues/routing-last-week-were-you.spec.js]: This test description doesn't match up to the test - it doesn't check that you are routed to question 35, it checks that you are _not_ routed to question 29.

File: tests/functional/spec/census/issues/routing-last-week-were-you.spec.js [hunk 3]
```diff
@@ -88,4 +89,98 @@ describe('ArriveInUk', function () {
     EmploymentType.submit()
     expect(Jobseeker.isOpen()).to.be.true
   })
+
+  it('Given I am answering question 28.	Last week were you.., When I select multiple answers including none of the above, Then I am routed to 29. Were you actively looking for...?', function () {
```
Comment (ajmaddaford) [inline tests/functional/spec/census/issues/routing-last-week-were-you.spec.js]: I think the description is wrong, the test checks that you are _not_ routed to question 29.

**Review by ajmaddaford — CHANGES_REQUESTED** [review]
I think there should also be a test for what should happen when none of the above only is selected. 
Also, can we add this to the routing tests rather than the issues?

**Review by ajmaddaford — APPROVED** [review]

File: tests/functional/spec/census/routing/last-week-were-you.spec.js [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+import Volunteering from '../../../pages/surveys/census/household/volunteering.page.js'
+import EmploymentType from '../../../pages/surveys/census/household/employment-type.page.js'
+import Jobseeker from '../../../pages/surveys/census/household/jobseeker.page.js'
+import MainJob from '../../../pages/surveys/census/household/main-job.page'
+
+const expect = chai.expect
+
+describe('Routing on Multiple Checkbox', function () {
+
+  it('Given I am answering question 28.	Last week were you.., When I dont not select any response, Then I am routed to 29. Were you actively looking for...?', function () {
```
Comment (samiwel) [inline tests/functional/spec/census/routing/last-week-were-you.spec.js]: dont not

**Review by samiwel — APPROVED** [review]
Looks okay. Just a general comment (not specifically to do with this PR) I'm finding it increasingly difficult to spot the "When" action in the Census selenium tests.

In future can we maybe add a comments such as
`// Given`, `// When`, `// Then` in the spec files just to make it a bit more obvious what's being tested?

File: tests/functional/spec/census/routing/last-week-were-you.spec.js (hunk 4 above)
Comment (ajmaddaford) [reply tests/functional/spec/census/routing/last-week-were-you.spec.js]: Also just spotted the weird spacing

Comment (jonnyshaw89) [reply tests/functional/spec/census/routing/last-week-were-you.spec.js]: This spacing doesn't appear in my IDE

Comment (samiwel) [reply tests/functional/spec/census/routing/last-week-were-you.spec.js]: I spotted that, but at least it was consistent across all tests.

**Comment by yoganandkunche** [comment]
Defect passed retest following deployment on 20th Dec 2016
